### ISSUE TITLE
Hotel for proxying from localnet

### ISSUE THREAD

**Issue by GomZik** [typicode/hotel#102]
If i run project inside vagrant or something like, it would be nice, if I can use hotel to proxying from localnte address (e.g. 192.168.56.101:8080).

Also, I have a question. Can I use just localhost address (e.g. localhost:8080), not an command as a server

**Comment by typicode** [comment]
Not yet but it's clearly on the roadmap. I'm thinking of something like:

```sh
hotel add 92.168.56.101:8080 -n app1
hotel add localhost:8080 -n app2
...
```

Would that do the trick for your setup?

**Comment by GomZik** [comment]
Yeah! It would be exactly, what i need! Hotel supports any type of connections (e.g. Websockets?) or only http(s)?

**Comment by typicode** [comment]
Any type of connections (ws, http and https).

**Comment by obezuk** [comment]
This would be an awesome addition. Looking forward to this feature :)

**Comment by emilniklas** [comment]
I'm also looking forward to this. It would really complete the coverage of my dev setup. A small side note:

@typicode 
> Any type of connections (ws, http and https).

If you're on https, you should use wss for web sockets. If this has been overlooked, I though I should mention it.

Great work with this project, it really works for me.

**Comment by typicode** [comment]
Work in progress ;) still need to figure out some things

![screenshot_2016-06-07](https://cloud.githubusercontent.com/assets/5502029/15854899/b118d072-2cac-11e6-9d6d-1db0bb558a1d.png)

> If you're on https, you should use wss for web sockets. If this has been overlooked, I thought I should mention it.

My bad, forgot to mention `wss`, all 4 types of connections will be supported (ws, wss, http and https) . Using `ws://` or `wss://` will need to be done on the front-end though.

> Great work with this project, it really works for me.

Thank you for the feedback and thoughts, I appreciate :)

**Comment by typicode** [comment]
Hey everyone,

I've published `v0.5.0-beta.2` with "proxying support". For the moment, it's only available using the following command `npm install -g hotel@next` (`README` is in the `next` branch).

To map a local domain to a URL, run:

```
hotel add http://... -n name
hotel add https://... -n name
```

Since it's a pretty important refactoring, I'd like to get user feedbacks first before making it available to everyone. If you install it let me know if it works for you or if you find some issues :)

**Comment by emilniklas** [comment]
I can't get past the arg parser:

```shell
> hotel add http://192.168.10.33
```

```
Usage: /usr/local/bin/hotel <command> [options]

Commands:
  add [-n name] [-o file] [-e env] [-p port] <cmd|url>      Add server or proxy
  rm [-n name]                                              Remove server or proxy
  ls                                                        List servers
  start                                                     Start daemon
  stop                                                      Stop daemon

Options:
  -v, --version  Show version number                                   [boolean]
  -h, --help     Show help                                             [boolean]

Examples:
  /usr/local/bin/hotel add nodemon
  /usr/local/bin/hotel add -o app.log 'serve -p $PORT'
  /usr/local/bin/hotel add -n app 'serve -p $PORT'
  /usr/local/bin/hotel add -e PATH 'serve -p $PORT'
  /usr/local/bin/hotel add -n app http://192.168.1.10

https://github.com/typicode/hotel

Not enough non-option arguments: got 1, need at least 5
```

**Comment by typicode** [comment]
Thanks for the report! I guess that's the definition of failing fast 😅 
I'll see if it's possible to update automated tests to use the arg parser instead of the CLI lib (which makes tests pass but in real life it doesn't work).

**Comment by typicode** [comment]
I've released a new beta with a working CLI, if you want to try `npm install -g hotel@next`.
The tests have been updated to use the CLI arg parser.

I've fixed also a lot of other small bugs. It should be better, but I'm sure there are still some left.

**Comment by emilniklas** [comment]
It works great!

One thing though. When the proxy request fails (i.e. my dev server is down) the daemon crashes, and I have to restart it. Would be cool if hotel could wait and poll the proxy target until it answers. That way, instead of hitting Cmd+R a dozen times until nodemon has restarted my server, it would just wait until the server responded. I'm not entirely sure how you have implemented the proxy, but it could be possible.

What do you think, @typicode?

**Comment by typicode** [comment]
Right now, I think the behaviour should be the following one:

__Command mapping__

```
hotel add nodemon --name app --port 4000
```

`GET http://app.dev` should start `nodemon`, poll `http://localhost:4000` for 5 seconds until it answers. If it answers the request is proxied else a `502` should be returned.

__URL mapping__

```
hotel add http://localhost:5000 --name app
```

`GET http://app.dev` should simply proxy request. I don't want to make this part too clever, it should be just URL mapping.

But since you're talking about `nodemon`, I'm not sure which case you're referring to. If you've done `hotel add nodemon` and it doesn't poll then it's a bug.

Also the daemon should never crash, do you still have the stacktrace in `~/.hotel/daemon.log`?

**Comment by emilniklas** [comment]
I'm using Vagrant and nodemon is running inside. Hotel runs on my host machine, so I needed the proxy to forward to my vm's IP.

I get why you wouldn't want too much functionality in the proxy, that's fine. 😊

**Comment by typicode** [comment]
I see, makes sense :)

**Comment by typicode** [comment]
Released `v0.5.0` :)
